apollo-federation/src/sources/connect/models.rs

```diff
@@ -227,21 +291,26 @@ impl HttpJsonTransport {
     }
 
     fn variables(&self) -> impl Iterator<Item = Namespace> + '_ {
-        let url_variables = self
-            .connect_template
-            .variables()
-            .map(|v| v.namespace.namespace);
+        self.variable_references()
+            .map(|var_ref| var_ref.namespace.namespace)
+    }
+
+    fn variable_references(&self) -> impl Iterator<Item = VariableReference<Namespace>> + '_ {
+        let url_variables = self.connect_template.variables().map(Clone::clone);
```

Comment:
[Iterator::cloned](https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.cloned)

```suggestion
        let url_variables = self.connect_template.variables().cloned();
```